### PR TITLE
Feature/47 first unused empty sequence

### LINKED ISSUE #47: Ctrl+Shift+F to insert first unused empty sequence
This is an old task imported from a `notes.txt` file where it was set as a pending hotkey.

An empty sequence has no data and is only one row in length.

### PR DESCRIPTION
**Pull request by micheldebree**
**Work in progress DO NOT MERGE**

TODO:
- [x] deduplicate code
- [ ] add to manual

P.S. 
- I seem to have removed trailing whitespace, please read through that
- I have included my autoformatting config aswell
- The tabs seem to really mess up the layout in Github, can we switch to spaces?

Resolves #47

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,139 @@
+---
```
Comment (micheldebree) [inline .clang-format:1]: This is my auto formatting config. Haven't used it yet on this code.

File: SIDFactoryII/source/runtime/editor/components/component_track.cpp [hunk 2]
```diff
@@ -130,10 +132,10 @@ namespace Editor
 		UpdateMaxEventPos();
 
 		m_FocusRow = ComponentTrackUtils::CalculateFocusRow(m_Dimensions.m_Height >> 1, m_Dimensions.m_Height);
-	
+
```
Comment (micheldebree) [inline SIDFactoryII/source/runtime/editor/components/component_track.cpp:135]: Removed all the trailing whitespace, sorry for the mess

File: SIDFactoryII/source/runtime/editor/components/component_track.cpp [hunk 3]
```diff
@@ -1881,6 +1883,40 @@ namespace Editor
 		return m_EventPos;
 	}
 
+	// TODO: deduplicate code, this is almost the same as DoInsertFirstFreeSequence
```
Comment (micheldebree) [inline SIDFactoryII/source/runtime/editor/components/component_track.cpp:1886]: Maybe parameterize the function? Only difference is the function that gets the sequence index.

File: SIDFactoryII/source/runtime/editor/components/component_track.cpp (hunk 2 above)
Comment (TheRealLaxity) [reply SIDFactoryII/source/runtime/editor/components/component_track.cpp:135]: I think the getfirstfree and getfirstunused sequence functions do not belong in the component base. I don't quite know why I put the firstfree one there in the first place.

File: SIDFactoryII/source/runtime/editor/components/component_track.cpp (hunk 3 above)
Comment (TheRealLaxity) [reply SIDFactoryII/source/runtime/editor/components/component_track.cpp:1886]: Agree

**Review by TheRealLaxity — APPROVED** [review]
I approve of the CL, however think that we need to look at those functions passed to the component base.

**Comment by TheRealLaxity** [comment]
And I just realized I misread something. Ignore my comments about the component base class. :)

File: SIDFactoryII/source/runtime/editor/components/component_track.cpp (hunk 3 above)
Comment (micheldebree) [reply SIDFactoryII/source/runtime/editor/components/component_track.cpp:1886]: Done. In the other branch..

**Comment by micheldebree** [comment]
Closed this PR because this branch is merged in branch #60 to make things simpler.

**Pull request closed without merging**